client/src/lib.rs

```diff
@@ -157,17 +156,27 @@ pub async fn aggregator_hpke_config(
         hpke_config_response.bytes().await?.as_ref(),
     ))?;
 
-    // TODO(#857): Pick one of the advertised HPKE configs. For now, just take the first one, since
-    // we support any HpkeConfig we can decode, and it should be the server's preferred one.
-    let hpke_config = hpke_configs
-        .hpke_configs()
-        .get(0)
-        .ok_or(Error::UnexpectedServerResponse(
+    if hpke_configs.hpke_configs().is_empty() {
+        return Err(Error::UnexpectedServerResponse(
             "aggregator provided empty HpkeConfigList",
-        ))?
-        .clone();
+        ));
+    }
 
-    Ok(hpke_config)
+    // Take the first supported HpkeConfig from the list. Return the first error otherwise.
+    let mut first_error = None;
+    for config in hpke_configs.hpke_configs() {
+        match is_hpke_config_supported(config) {
+            Ok(()) => return Ok(config.clone()),
+            Err(e) => {
+                if first_error.is_none() {
+                    first_error = Some(e);
+                }
+            }
+        }
```

Comment:
```suggestion
            Err(e) => first_error.get_or_insert(e),
        };
```
nit, can take it or leave it: [`get_or_insert()`](https://doc.rust-lang.org/std/option/enum.Option.html#method.get_or_insert) inserts when the contained value is `None`. Might be dubious whether this is a readability improvement though.

Reply:
Hmm, I think I'll pass, since we're not using the `&mut T` it returns in this case, but just the side-effect.